functionsLab.js

```diff
@@ -0,0 +1,296 @@
+("----------------------------------------------------");
+// ## Question 7
+//
+// Write a function named `first` that takes a number and returns an array with all the numbers from 1 to n
+//
+//
+// Example:
+//
+// Input: `first(3)`
+//
+// Output: `[1, 2, 3]`
+const first = (num) => {
+  let tempArray = [];
+  for(let i = num; (i <= num) && (i >= 1); i--) {
+    tempArray.push(i)
+  }
+  for(let i = 0; i < tempArray.length; i++) {
+    console.log(tempArray[i]);
+  }
+}
+first(5);
+("----------------------------------------------------");
+// ## Question 8
+//
+// Write a function that logs the numbers from 1 to x, except:
+//
+// If the number if a multiple of 3, log `"Fizz"` instead of the number
+// If the number is a multiple of 5, log `"Buzz"` instead of the number
+// If the number is a multiple of 3 AND 5, log `"FizzBuzz"` instead of the number
+// Your function should take in one parameter: x (the number to count up to)
+let function8 = (num) => {
+  for(let i = 1; i < num; i++) {
+    if( (i % 3 === 0) && (i % 5 === 0) ) {
+      console.log("FizzBuzz");
+    } else if(i % 3 === 0) {
+      console.log("Fizz");
+    } else if(i % 5 === 0) {
+      console.log("Buzz");
+    } else {
+      console.log(i);
+    }
+  }
+}
+function8(100);
+console.log("----------------------------------------------------");
+// ## Question 9
+//
+// Write a function named `reverse` that takes an array as a parameter. The function should return an array in reverse order.
+//
+//
+// Example:
+//
+// Input: `reverse([1, 2, 3])`
+//
+// Output: `[3, 2, 1]`
+const reverse = (arrayInput) => {
+  let newArr = [];
+  for(let i = arrayInput.length-1; i >= 0 ; i--) {
+    newArr.push(arrayInput[i]);
+  }
+  return newArr;
+  }
+
+let array9 = [0, 1, 2, 3, 4];
+console.log(reverse(array9));
+console.log("----------------------------------------------------");
+// ## Question 10
+//
+// Write a function that prints out the most frequently occurring element in an array.
+//
+// Example:
+//
+// Input: `[2,1,4,2,5,2,3,23,3,3,2,2,2,24,56,2,2,5,46,2]`
+//
+// Output: `2`
+// const function10 = (inputArray, n) => {
+//   inputArray.sort();
+//   let max_count = 1;
+//   let res = inputArray[0];
+//   let curr_count = 1;
+//   for (let i = 0; i < n; i++) {
+//     if (inputArray[i] == inputArray[i - 1]) {
+//         curr_count++;
+//       } else {
+//         if (curr_count > max_count)
+//         {
+//             max_count = curr_count;
+//             res = inputArray[i - 1];
+//         }
+//         curr_count = 1;
+//     }
+//   }
+//   // If last element is most frequent
+//           if (curr_count > max_count)
+//           {
+//               max_count = curr_count;
+//               res = inputArray[n - 1];
+//           }
+//
+//           return res;
+// }
+// let array10 = [2, 1, 4, 2, 5, 2, 3, 23, 3, 3, 2, 2, 2, 24, 56, 2, 2, 5, 46, 2];
+// let n = array10.length;
+// function10(array10);
+//or I can do this way
+const function10 = () => {
```

Comment:
This question is only answered half way since we visually have to determine which number is repeated the most. What we want to do is to make our program print what the answer is. Seeing which number is repeated the most manually would be tough if we are dealing with millions of numbers. How can we solve the other half?